doc/dev-guide.md

```diff
@@ -458,6 +458,28 @@ sqldb.query(sql.select_questions, params, ...);
 SELECT * FROM questions WHERE id IN (SELECT unnest($id_list::INTEGER[]));
 ```
 
+* To pass a lot of data to SQL a useful pattern is to send a JSON object array and unpack it in SQL to the equivalent of a table. This is the pattern used by the "sync" code, such as [sprocs/sync_news_items.sql](https://github.com/PrairieLearn/PrairieLearn/blob/master/sprocs/sync_news_items.sql). For example, given the JavaScript:
+```javascript
+let data = [
+    {a: 5, b: "foo"},
+    {a: 9, b: "bar"}
+];
+let params = {data};
```

Comment:
This is cool, didn’t know this syntax was a thing to create both a key and value!

Reply:
It's needed for the `WITH ORDINALITY`. I split this out as a second bullet point to make it clearer.